src/main/java/org/opensearch/security/configuration/SecurityConfigVersionHandler.java

```diff
@@ -0,0 +1,360 @@
+                                                                                                // manager node
+
+        if (!isVersionIndexEnabled(settings)) return;
+
+        try {
+            log.info("Initializing version index ({})", SecurityConfigVersionsIndex);
+
+            if (!createOpendistroSecurityConfigVersionsIndexIfAbsent()) {
+                log.info("Version index already exists, skipping initialization.");
+            }
+
+            waitForOpendistroSecurityConfigVersionsIndexToBeAtLeastYellow();
+
+            String nextVersionId = fetchNextVersionId();
+            User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+            String userinfo = (user != null) ? user.getName() : "system";
+
+            Version<?> version = buildVersionFromSecurityIndex(nextVersionId, userinfo);
+            saveCurrentVersionToSystemIndex(version);
+
+        } catch (Exception e) {
+            log.error("Failed to initialize config version index", e);
+        }
+    }
+
+    boolean createOpendistroSecurityConfigVersionsIndexIfAbsent() {
+        try {
+            final Map<String, Object> indexSettings = ImmutableMap.of("index.number_of_shards", 1, "index.auto_expand_replicas", "0-all");
+
+            final Map<String, Object> mappings = Map.of(
+                "properties",
+                Map.of(
+                    "versions",
+                    Map.of(
+                        "type",
+                        "object",
+                        "properties",
+                        Map.of(
+                            "version_id",
+                            Map.of("type", "keyword"),
+                            "timestamp",
+                            Map.of("type", "date"),
+                            "modified_by",
+                            Map.of("type", "keyword"),
+                            "security_configs",
+                            Map.of("type", "object", "enabled", false)
+                        )
+                    )
+                )
+            );
+            log.info("Index request for {}", SecurityConfigVersionsIndex);
+            final CreateIndexRequest createIndexRequest = new CreateIndexRequest(SecurityConfigVersionsIndex).settings(indexSettings)
+                .mapping(mappings);
+
+            final boolean ok = client.admin().indices().create(createIndexRequest).actionGet().isAcknowledged();
+            log.info("Index {} created?: {}", SecurityConfigVersionsIndex, ok);
+            return ok;
+        } catch (ResourceAlreadyExistsException resourceAlreadyExistsException) {
+            log.info("Index {} already exists", SecurityConfigVersionsIndex);
+            return false;
+        } catch (Exception e) {
+            log.error("Failed to create index {}", SecurityConfigVersionsIndex, e);
+            throw e;
+        }
+    }
+
+    void waitForOpendistroSecurityConfigVersionsIndexToBeAtLeastYellow() {
```

Comment:
Can we abstract this to a public static method that can be re-used? This looks identical to `ConfigurationRepository.waitForSecurityIndexToBeAtLeastYellow`

Reply:
@divyanshi-0402 The ask is to take the common code from both the classes and put in one method to avoid duplication.